### DISCUSSION TITLE
LogB min & max values

### DISCUSSION THREAD

**Discussion by RikshaDriver** [gopro/labs#145] [Q&A]
With the new LogB custom functions, what are the absolute minimum and maximum logbase and black offset values we can use?

**Comment by dnewman-gpsw** [comment]
The linked Log Curve Designer has the same math as used in Labs firmware: https://www.desmos.com/calculator/sfxqtunzty use to study what the curve will be with oMLOGB=b,o inputs. Only difference is you can go more than 4000 log on the camera. Log base supports 2 to 1000000, although useful range is more like 100-10000. Black offset supports -100 to 500.  The minus range add a black offset (and peak white offset.) Some older video tools can't handle full range (0-1023), so using a black offset of -64, will get you 64-960 range.  This is the deepest of hacks for Labs to-date, so please let me know how you end up using it, and what input values you find useful.

**Comment by RikshaDriver** [comment]
Thanks.

Would be great to expand for x values beyond 1 to truly take advantage of the Dynamic Range.

Limiting it to x<=1 simply alters the placement of the code values that are used, but the Dynamic Range is still limited to SDR Peak White.

I currently under-expose Flat by ~2 stops then adjust in post to get the extra range, but with these new settings...

One could for example use b = 1000000 and o = -128 and if input is allowed beyond 1 to 5.6... this would theoretically allow 4+ stops above mid-gray.

**Reply by dnewman-gpsw** [reply]
You should be combining with EV, as 1.0 is the chipping point, it would be if there was not matter the log curve. All sensors clip at 1.0, it is up to you to choice a non-1.0 white point.  Using EV -2 and a more pronounced Log curve would give a little more in dynamic range. This is still a small pixel you can't gain a crazy amount of DR, but do to see what you can get.

**Reply by RikshaDriver** [reply]
Thanks for clarification that sensor clipping is mapped to 1.0

I've managed decent DR with -2 EV using the Flat profile.

Will give the custom settings a go if I can I get my hands on the 11.
